public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -126,7 +126,8 @@ export class PanelModel {
 
   ensureQueryIds() {
     if (this.targets) {
-      for (const query of this.targets) {
+      for (let i = 0; i < this.targets.length; i++) {
+        const query = this.targets[i];
```

Comment:
this fix feels fragile, especially without a test. Maybe check if it's an array instead and keep the other for of loop. Not sure why this makes a difference, for of loops generate a for loop like above so this fix should not make a difference.

Reply:
The problem is that in my case targets contains some artifacts - it's array-like object with additional props:

![Screenshot from 2019-03-15 10-22-29](https://user-images.githubusercontent.com/4932851/54415093-4c7dc000-470c-11e9-93ee-51884303b988.png)

`for ... of` returns first element of array (I think it treats my object as array with length 3) and then returns `undefined` on 2nd and 3rd iterations.

Reply:
I tried to use 
```js
      for (const query of this.targets) {
        if (query && !query.refId) {
          query.refId = this.getNextQueryLetter();
        }
      }
```
but it causes infinite loop for my object.

Reply:
This approach works as well:
```ts
  ensureQueryIds() {
    if (this.targets && _.isArray(this.targets)) {
      for (const query of this.targets) {
        if (!query.refId) {
          query.refId = this.getNextQueryLetter();
        }
      }
    }
  }
```